File: surix-frontend/src/types/producto.ts
export type Producto = {
  id: number
  nombre: string
  precio: number
  descripcion: string | null
  stock: number
  estanteId: number | null
  estanteNombre: string | null
  ordenLogico: number | null
  categoriaId: number | null
  categoriaNombre: string | null
}

export type ProductoFormData = {
  nombre: string
  precio: number
  stock: number
  descripcion?: string
  estanteId: number
  categoriaId: number
}

export type ProductoUpdateData = Partial<ProductoFormData>

File: backend-spring/src/main/java/com/surixapp/mercado/entity/Producto.java
package com.surixapp.mercado.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "productos")
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 150)
    private String nombre;

    @Column(nullable = false)
    private Double precio;

    @Column(nullable = false)
    private Integer stock = 0;

    @Column(length = 500)
    private String descripcion;

    // ManyToOne con Estante
    @ManyToOne
    @JoinColumn(name = "estante_id")
    private Estante estante;

    @ManyToOne
    @JoinColumn(name = "categoria_id")
    private Categoria categoria;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getNombre() { return nombre; }
    public void setNombre(String nombre) { this.nombre = nombre; }
    public Double getPrecio() { return precio; }
    public void setPrecio(Double precio) { this.precio = precio; }
    public Estante getEstante() { return estante; }
    public void setEstante(Estante estante) { this.estante = estante; }
    public Categoria getCategoria() { return categoria; }
    public void setCategoria(Categoria categoria) { this.categoria = categoria; }
    public String getDescripcion() { return descripcion; }
    public void setDescripcion(String descripcion) { this.descripcion = descripcion; }
    public Integer getStock() { return stock; }
    public void setStock(Integer stock) { this.stock = stock; }
    
}